pkg/controller/mcpserver_controller.go

```diff
@@ -223,16 +232,17 @@ func (r *MCPServerReconciler) checkReadyCondition(ctx context.Context, server *k
 	}
 
 	// Check if deployment is available
-	if deployment.Status.ReadyReplicas > 0 && deployment.Status.ReadyReplicas == deployment.Status.Replicas {
+	// A deployment is considered ready when it has the desired number of available replicas
+	if deployment.Status.AvailableReplicas > 0 && deployment.Status.AvailableReplicas == deployment.Status.Replicas {
 		setReadyCondition(
 			server,
 			true,
 			kagentdevv1alpha1.MCPServerReasonReady,
```

Comment:
nit: we should add the `MCPServerReasonAvailable` and `MCPServerReasonNotAvailable` constants and update them here.